### ISSUE TITLE
Unit Test Coverage

### ISSUE THREAD

**Issue by paddatrapper** [Progether/JAdventure#196]
Tests should cover the majority of the code base. Thus [code coverage](http://progether.github.io/JAdventure/jacoco/index.html) should be over 80%. Currently it is sitting at 25%

**Comment by nicce** [comment]
I'll happily take a look at this one. Not sure if a percentage limit is what to aim after. However I add test cases to where I see fit, and then you can tell me if you think I have missed something.

**Comment by paddatrapper** [comment]
Yeah, I agree with you. Thanks for looking into this!

**Comment by Ronan-H** [comment]
Should the classes in *com.jadventure.game.menus* be unit tested? I see there are currently no unit tests at all for those classes.

**Comment by paddatrapper** [comment]
Ideally yes, but I can't remember if there was an issue with it or if it just hasn't happened yet

**Comment by Ronan-H** [comment]
@paddatrapper Having a look right now. What would be the best way to test input/output to the menus? I see CommandCollectionTest uses _System.setOut_ to test the output. Would it be OK to use _System.setIn_ and _System.setOut_, and call _QueueProvider.startMessenger(GameModeType.STAND_ALONE)_ to test the menus?

**Comment by paddatrapper** [comment]
@Ronan-H yeah that is probably the best way to do it. You may be able to tie into the `QueueProvider`, but it may well not be possible.

**Comment by Ronan-H** [comment]
Opened an issue about a rework idea for QueueProvider in #258 to make testing easier.
